File: solutions/csharp/cricinfo/Innings.cs
using System.Collections.Generic;

namespace Cricinfo
{
    public class Innings
    {
        private readonly string id;
        private readonly string battingTeamId;
        private readonly string bowlingTeamId;
        private readonly List<Over> overs;

        public Innings(string id, string battingTeamId, string bowlingTeamId)
        {
            this.id = id;
            this.battingTeamId = battingTeamId;
            this.bowlingTeamId = bowlingTeamId;
            this.overs = new List<Over>();
        }

        public void AddOver(Over over)
        {
            this.overs.Add(over);
        }

        public string GetId()
        {
            return id;
        }

        public List<Over> GetOvers()
        {
            return overs;
        }

        public string GetBattingTeamId()
        {
            return battingTeamId;
        }

        public string GetBowlingTeamId()
        {
            return bowlingTeamId;
        }
    }
}

File: solutions/java/src/cricinfo/Innings.java
package cricinfo;

import java.util.ArrayList;
import java.util.List;

public class Innings {
    private final String id;
    private final String battingTeamId;
    private final String bowlingTeamId;
    private final List<Over> overs;

    public Innings(String id, String battingTeamId, String bowlingTeamId) {
        this.id = id;
        this.battingTeamId = battingTeamId;
        this.bowlingTeamId = bowlingTeamId;
        this.overs = new ArrayList<>();
    }

    public void addOver(Over over) {
        this.overs.add(over);
    }

    public String getId() {
        return id;
    }

    public List<Over> getOvers() {
        return overs;
    }

    public String getBattingTeamId() {
        return battingTeamId;
    }

    public String getBowlingTeamId() {
        return bowlingTeamId;
    }
}
